### PR TITLE
Updates from SAML SSO - product documentation for NOV-22 release

### PR DESCRIPTION
**Pull request by MarkvanMents**

### CODE REVIEW

File: content/en/docs/appstore/modules/saml.md [hunk 1]
```diff
@@ -136,15 +138,15 @@ To configure SAML, start your app and navigate to the **OpenConfiguration** micr
 
 You can now finish configuring your SAML module in your app by reviewing/updating the Service Provider (SP), and creating/updating the IdP configuration.
 
-### 3.1 Configuring Service Provider
+### 3.1 Configuring Service Provider{configure-sp}
```
Comment (MarkvanMents) [inline content/en/docs/appstore/modules/saml.md:141]: Are these in the right order?
Phanindra Babu Chennu  [11 hours ago](https://paper.dropbox.com/doc/1nREI0KMpv0qw5ii0AipI?t=039914464272796124200840#:t=039914464272796124200840)
​​SAML Module will add IDP alias name to the existing URL based on the respective IDP configuration
JF
Jaap Francke  [3 hours ago](https://paper.dropbox.com/doc/1nREI0KMpv0qw5ii0AipI?t=039914464272796124200840#:t=039914464272796124200840)
​​Understood. It means that as an admin you would first have to do the IDP configuration (section 3.2) and after that do the SP configuration (section 3.1). Agree? This document says: first configure SP, and then configure the IDP. So I thik that’s wrong. Agree @phani/ ​[@Naveen C](https://paper.dropbox.com/ep/profile/1gg9uA1iUgdDmqkTo2qowQVjNd0Rk4pW5ITZqT6JVvSAW6P0kuun)​ ?

File: content/en/docs/appstore/modules/saml.md [hunk 2]
```diff
@@ -155,9 +157,9 @@ You can choose what you want to enter for the **Entity Id**, **Organization**, a
...
+    If encryption is enabled, all the certificates required for encryption are stored in the key store. When you choose **Use encryption** a key store is automatically created using the URL of the application, or the custom EntityID, and shown as the **Key store alias**. See [Managing the Key Store](#keystore), below, for more key store options.
 
-Accessing the metadata for the SP can be done either by clicking **Download SP Metadata** to download the XML file or by opening `http://<Application Root URL>/SSO/metadata` for your app's URL.
+The SAML module generates separate SP metadata for every connected SAML IDP (see [Configuring the IdP-Specific Settings](#idp-specific-settings)).  The SP metadata for your app can be obtained by clicking **Download SP Metadata** to download the XML file or by opening `http://<Application Root URL>/SSO/metadata` for your app's URL.
```
Comment (MarkvanMents) [inline content/en/docs/appstore/modules/saml.md:164]: There seems to be no documentation of the buttons in this image. What do **Previous** and **Save** do? What happens if you click **Cancel**?
image

**Review by MarkvanMents — COMMENTED** [review]
Looks good. I have two outstanding questions.

File: content/en/docs/appstore/modules/saml.md (hunk 1 above)
Comment (naveen-chikurthi) [reply content/en/docs/appstore/modules/saml.md:141]: Hi Mark, yes it is the correct order. First  SP needs to be configured, then only we can be able to configure IDP.

File: content/en/docs/appstore/modules/saml.md (hunk 2 above)
Comment (naveen-chikurthi) [reply content/en/docs/appstore/modules/saml.md:164]: Hi Mark, The below are the actions that buttons do:

- **Previous**: This button will navigate user to the previous step in dialog of the configuration
- **Save**: This button will save the IDP configuration and closes the dialog
- **Cancel** : This just a Mendix dialog cancel action which closes the dialog

File: content/en/docs/appstore/modules/saml.md (hunk 1 above)
Comment (MarkvanMents) [reply content/en/docs/appstore/modules/saml.md:141]: Great - I'll leave it as it is then.

File: content/en/docs/appstore/modules/saml.md (hunk 2 above)
Comment (MarkvanMents) [reply content/en/docs/appstore/modules/saml.md:164]: I'll just add that to the text then.
Thanks for the response.

**Review by MarkvanMents — COMMENTED** [review]
Responded to @naveen-chikurthi 's replies.

File: content/en/docs/appstore/modules/saml.md [hunk 3]
```diff
@@ -136,16 +138,17 @@ To configure SAML, start your app and navigate to the **OpenConfiguration** micr
 
 You can now finish configuring your SAML module in your app by reviewing/updating the Service Provider (SP), and creating/updating the IdP configuration.
 
-### 3.1 Configuring Service Provider
+### 3.1 Configuring Service Provider{configure-sp}
 
 Before any IdP can be configured, you need to configure the SP, which is your current application. The SP configuration allows you to configure some basic information for the SP metadata file. This information will be be available in the IdP for the reference of the IdP administrator.
 
+You will need to fill in several dialog boxes of information. Use **Previous** to go back to the previous dialog, or **Cancel** to abandon your changes.
```
Comment (MarkvanMents) [inline content/en/docs/appstore/modules/saml.md]: Hi @naveen-chikurthi 
I've added this information. Does this section look OK with the new SP configuration flow, or does it need updating. I don't have the latest version of the module so I can't see how it works.

**Comment by MarkvanMents** [comment]
OK to publish when requested.

File: content/en/docs/appstore/modules/saml.md (hunk 3 above)
Comment (naveen-chikurthi) [reply content/en/docs/appstore/modules/saml.md]: Hi @MarkvanMents, 
Current configuration scenario:
1. SP is configured directly through page, there are no dialogs in that configuration
2.IDP is being configured through Dialogs where we find "Previous" "Next" "SAVE" "Cancel" buttons

I think you have mentioned information about the buttons in the SP-metadata configuration section. Can it be moved to IDP configuration section?

Comment (MarkvanMents) [reply content/en/docs/appstore/modules/saml.md]: Will do that. Thanks for the information - and the short meeting.

**Comment by MarkvanMents** [comment]
@naveen-chikurthi 
I've made the changes we discussed in this commit. https://github.com/mendix/docs/pull/5347/commits/3b61f789eb2a8f9d68ff2548ed5075a6ac5d3a85

Can you check that it now makes sense?

Can you also have a look at section 6. This lists a whole load of URLs, but I think some of them may have changed because of the support for multiple IdPs. For example:
> * **/SSO/metadata** – This provides a point for the IdP to automatically download the metadata from this SP
should this also include the IDP-alias?

**Comment by naveen-chikurthi** [comment]
Hi @MarkvanMents , Yes. it would be better if we change /SSO/Metadata url by including IDP alias name as well (/SSO/metadata/<IDP-AliasName>)

**Comment by MarkvanMents** [comment]
> Hi @MarkvanMents , Yes. it would be better if we change /SSO/Metadata url by including IDP alias name as well (/SSO/metadata/)

Done that: https://github.com/mendix/docs/pull/5347/commits/3a2fcbe936dcaa65407dba13034f6d4c04893cdb

**Comment by MarkvanMents** [comment]
New SAML module release confirmed over Slack.

**Pull request merged**